grpc-sys/build.rs

```diff
@@ -12,14 +12,196 @@
 // limitations under the License.
 
 extern crate gcc;
+extern crate cmake;
+extern crate pkg_config;
+
+const GRPC_VERSION: &'static str = "1.3.0";
+
+#[cfg(not(feature = "static-link"))]
+mod imp {
+    use gcc;
+    use pkg_config;
+
+    use super::GRPC_VERSION;
+
+    pub fn build_or_link_grpc(cc: &mut gcc::Config) {
+        if let Ok(lib) = pkg_config::Config::new()
+               .atleast_version(GRPC_VERSION)
+               .probe("grpc_unsecure") {
+            for inc_path in lib.include_paths {
+                cc.include(inc_path);
+            }
+        } else {
+            panic!("can't find a dynamic grpc library");
+        }
+    }
+}
+
+#[cfg(feature = "static-link")]
+mod imp {
+    use std::env;
+    use std::fs::DirBuilder;
+    use std::process::Command;
+
+    use cmake;
+    use gcc;
+
+    use super::GRPC_VERSION;
+
+    const ZLIB_VERSION: &'static str = "1.2.8";
+    const BORINGSSL_COMMIT_HASH: &'static str = "78684e5b222645828ca302e56b40b9daff2b2d27";
+    const CARES_TAG: &'static str = "cares-1_12_0";
+
+    struct FakeGitSubmodule {
```

Comment:
Still here,  FakeGit confuses me, I suggest removing it or using another better name.